### ISSUE TITLE
[BUG] Datasets do not pass `min_tiles` filter when training GANs

### ISSUE THREAD

**Issue by siddhir** [jamesdolezal/slideflow#315]
## Description
GAN training io issue likely because it takes in the filters dictionary but not the .filter() function when constructing this initial dataset for io. Filtering after this step doesn’t seem to work and neither does including min_tiles=21 within the initial P.dataset(….min_tiles=21, filters={….}) call. Given I couldn’t figure out a way to include the min_tiles within the dictionary, I basically duplicated the annotation file and included another filter called “MIN_TILE” which excluded all WSIs with <21 extracted tiles. 

## To Reproduce
1. Use min_tiles argument when constructing any dataset for GAN

def main():
    P = sf.Project('/mnt/data/PROJECTS/Title')
    P.annotations = './qc_plan.csv'
    full_dataset = P.dataset(
    tile_px=299, 
    tile_um=302, 
    filters={'site': ['XXXXXXXXXXX], 
             'progressor':[XXXXX],
             'QC_Sid': ['QC_PASS'],
            },
    min_tiles=21
)

    train_df = full_dataset.filter({'site': [XXXXXXX]}) 

    P.gan_train(
        dataset=train_df,
        model='stylegan3',
        cfg='stylegan2',
        gpus=4,
        batch=128,
        kimg=10000,
        resize=256,
        tile_labels= "label",
        exp_label="label"
)

if __name__ == '__main__':
    mp.freeze_support()
    main()

## Expected behavior
Should take in the min_tiles argument when constructing dataset

## Environment:

 - Slideflow Version (e.g., 1.0):
 - OS (e.g., Ubuntu):
 - How you installed Slideflow (`pip`, source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by jamesdolezal** [comment]
This should be fixed in the latest `dev` update (f939885). Give it a try and see if it works - as a reminder, you'll need to update both the main slideflow repository as well as the GAN submodules:

```
git checkout dev 
git pull
git submodule update --recursive
```

**Comment by jamesdolezal** [comment]
Released in 2.2 - thanks!
